app/services/save_offer_conditions_from_params.rb

```diff
@@ -11,9 +11,13 @@ def save
     ActiveRecord::Base.transaction do
       @offer = Offer.find_or_create_by(application_choice: application_choice)
 
+      raise CannotEditNonPendingConditionsError if @offer.has_non_pending_conditions?
+
       serialize_standard_conditions
       serialize_further_conditions
     end
+  rescue CannotEditNonPendingConditionsError
```

Comment:
Not a blocker... Is there a reason for the `raise -> rescue -> raise` flow here? Can't we just raise `ValidationError` with the appropriate message within the transaction (or via a method)? 
I can understand defining a custom exception if the error propagates but in this case we're not seeing the custom exception outside of this service so I wonder if it's really necessary.

Reply:
Good point, will tidy up!